phys/module_ra_eclipse.F

```diff
@@ -0,0 +1,439 @@
+  integer,                             intent(in)    :: year
+! Geographical information
+  real,    dimension(ims:ime,jms:jme), intent(in)    :: xlat,xlon
+  real,    dimension(ims:ime,jms:jme), intent(inout) :: Obscur
+! Eclipse variables
+  integer, dimension(ims:ime,jms:jme), intent(inout) :: mask
+  real,                                intent(inout) :: elat_track,elon_track
+! Namelist option 1 - enabled / 0 - disabled
+  integer,                             intent(in)    :: sw_eclipse
+
+!----------
+! Local vars
+!----------
+!
+! Constants
+  real,    parameter       :: pi      = 3.1415926535897932384626433
+  real,    parameter       :: epsil   = 0.0818192 ! Earth excentricity
+!
+! Besseliam elements from file eclipse_besselian_elements.dat
+!	X, Y         :: coordinates of the eclipse's axis in the fundamental plane
+!       d, mu        :: declination and azimuth (deg)
+!       l1, l2       :: radii of the penumbra and umbra regions in the fundamental 
+!                       plane
+!	tanf1, tanf2 :: angle of the shadow's conus
+!       t0, min, max :: time when the Besselian elements are valid
+!       Dt           :: time correction
+  integer, parameter       :: NBessel = 4 ! Dimension
+  real, dimension(NBessel) :: X, Y, d, l1, l2, mu
+  real                     :: tanf1, tanf2, t0, tmin, tmax, Dt
+!
+! d and mu in radians
+  real                     :: rmu, rd
+!
+! Besselian elements at the current time t1
+  real                 :: cx, cy, cd, cl1, cl2, cmu
+!
+! Current time t1, and hourly angle H
+  real                 :: t1, &
+                           H   
+!
+! Latitude and longitude in radians
+  real                 :: rlat, rlon
+!
+!
+! Geocentric coordinates
+!----------
+  real                 :: cos_lat1, sin_lat1 ! cos/sin of the corrected latitude
+  real                 :: xi, eta1, zeta1      ! Geocentric coordinates at the observation point
+!
+! Eclipse variables
+  real                 :: DELTA, DELTA2      ! Distance of the observation point
+                                             ! to the eclipse axis
+  real                 :: LL1, LL2           ! Penumbra and Umbra radius
+                                             ! from the point observer's plane
+  logical              :: eclipse_exist
+!
+! Loop indeces
+  integer              :: i, j
+!
+! Internal variables
+  real                 :: A, B
+  real                 :: da,eot,xt24
+  integer              :: julday
+!
+! Eclipse path
+  real :: E, E_1, tan_rd1, rd1, rho1, cy1, tan_gamma, cgamma, sin_beta, &
+          tan_C, C, cm, tan_lat1, tan_lat, sin_H,                       &
+          cos_H, tan_H, beta, lon, tan_rd2, rd2, rho2
+!
+! Missing value
+  real :: MISSING
+
+
+!--------------
+!----------
+! CODE CUSTOMIZATION
+!----------
+!--------------
+!
+  MISSING = 1E30
+
+!--------------
+!----------
+! MAIN
+!----------
+!--------------
+!
+!----------
+! Check if the eclipse is enabled by the namelist.input
```

Comment:
@irowebbn 
Isaac,
This should probably not be in this main routine, but in some initialization routine that happens one time.